patches/net/minecraft/world/entity/LivingEntity.java.patch

```diff
@@ -132,36 +140,48 @@
          float f = this.getHealth();
          if (f > 0.0F) {
              this.setHealth(f + p_21116_);
-@@ -1118,6 +_,7 @@
- 
-     @Override
-     public boolean hurt(DamageSource p_21016_, float p_21017_) {
-+        if (!net.neoforged.neoforge.common.CommonHooks.onLivingAttack(this, p_21016_, p_21017_)) return false;
-         if (this.isInvulnerableTo(p_21016_)) {
+@@ -1127,23 +_,26 @@
+         } else if (p_21016_.is(DamageTypeTags.IS_FIRE) && this.hasEffect(MobEffects.FIRE_RESISTANCE)) {
              return false;
-         } else if (this.level().isClientSide) {
-@@ -1136,14 +_,17 @@
+         } else {
++            this.damageContainer = this.damageContainer == null ? new net.neoforged.neoforge.common.damagesource.DamageContainer.InternalDamageContainer(p_21016_, p_21017_) : this.damageContainer;
++            if (!net.neoforged.neoforge.common.CommonHooks.onEntityPreDamage(this, this.damageContainer)) return false;
+             if (this.isSleeping() && !this.level().isClientSide) {
+                 this.stopSleeping();
+             }
+ 
+             this.noActionTime = 0;
+-            float f = p_21017_;
++            float f = p_21017_ = this.damageContainer.getNewDamage();
              boolean flag = false;
              float f1 = 0.0F;
-             if (p_21017_ > 0.0F && this.isDamageSourceBlocked(p_21016_)) {
+-            if (p_21017_ > 0.0F && this.isDamageSourceBlocked(p_21016_)) {
 -                this.hurtCurrentlyUsedShield(p_21017_);
 -                f1 = p_21017_;
 -                p_21017_ = 0.0F;
-+                net.neoforged.neoforge.event.entity.living.ShieldBlockEvent ev = net.neoforged.neoforge.common.CommonHooks.onShieldBlock(this, p_21016_, p_21017_);
-+                if(!ev.isCanceled()) {
-+                if(ev.shieldTakesDamage()) this.hurtCurrentlyUsedShield(p_21017_);
++            net.neoforged.neoforge.event.entity.living.DamageBlockEvent ev;
++            if (p_21017_ > 0.0F && (ev = this.damageContainer.setBlockedDamage(net.neoforged.neoforge.common.CommonHooks.onDamageBlock(this, this.damageContainer, this.isDamageSourceBlocked(p_21016_)))).getBlocked()) {
```

Comment:
Is there a way to de-inline this statement?  This level of complexity in this single line makes it hard to read (which means it may be hard to maintain)

Reply:
I will break the update of the damage container to happen after the condition.  The rest should still remain in the condition.